ci/infra/testrunner/platforms/terraform.py

```diff
@@ -115,11 +115,18 @@ def _generate_tfvars_file(self):
             with open(tfvars_final, "w") as f:
                 json.dump(tfvars, f)
 
-    #take up to 50 characters from stackname to give room to the fixed part
-    # in the node name: caasp-[master|worker]-xxx (total length must be <= 63)
+    # take up to 45 characters from stackname to give room to the fixed part
+    # in the node name: caasp-[master|worker]-<stack name>-xxx (total length
+    # must be <= 63).
+    # Also ensure that only valid character are present and that the string
+    # starts and ends with alphanumeric characters and all lowercase.
     def stack_name(self):
-         stack_name = self.conf.terraform.stack_name
-         return stack_name[:45].lower()
+         stack_name = self.conf.terraform.stack_name[:45]
+         stack_name = stack_name.replace("_","-").replace("/","-")
+         stack_name = stack_name.strip("-.")
```

Comment:
The comment says this ensures that only valid characters are present and that it starts/ends with alnum.  But the code says "remove underscores and slashes, and make sure it doesn't start or end with a period or hyphen."

Why doesn't the code do what the comment says?  Is there a guarantee elsewhere that underscore, slash, and period are the only non-alnum chars which could possibly be in this value?

Reply:
I'd personally be _strongly_ inclined to use something like `stack_name = re.sub(r'[^.A-Za-z\d]', '-', stack_name)` to replace all chars that aren't period or alnum with a hyphen.

Reply:
Hi Dany. I agree that this is the safest solution. I was working on the assumption that the problem was detected in the context of the ci, where the only possible invalid character is "_". But your suggestion would be a good follow up to this. Thanks